src/jv.c

```diff
@@ -1781,6 +1791,7 @@ jv jv_object_delete(jv object, jv key) {
   assert(JVP_HAS_KIND(key, JV_KIND_STRING));
   jvp_object_delete(&object, key);
   jv_free(key);
+  if(jv_is_borrowed(object)) return jv_unborrow(object);
```

Comment:
I think we need to do the unborrow before the `jvp_object_delete()`, no?  Because we're modifying the `object`, therefore we must own the reference.

Reply:
indeed it fails

Reply:
could jv_array_slice() cause the same issue?

Reply:
nvm added a test for it and it passes just fine

Reply:
jv_object_set() should be problematic as well

Reply:
I've now gone through the entire jv api (at least every function I deemed relevant on https://github.com/jqlang/jq/wiki/C-API:-jv)